Added tests for dvr router type change

TregubovYury opened the pull request:
Author: Tregubov Yury

Added tests for QA-311 QA-312 QA-313 QA-315

Also updated TestRail ids for my previous test in the ovs restart

rpromyshlennikov committed:
Test ban dhcp agent refactoring (new base class).
Typo fixes.

Commit:
Added tests for dvr router type change

Author: Tregubov Yury

Added tests for QA-311 QA-312 QA-313 QA-315

Also updated TestRail ids for my previous test in the ovs restart

Commit:
Added tests for dvr router type change

Author: Tregubov Yury

Added tests for QA-311 QA-312 QA-313 QA-315

Also updated TestRail ids for my previous test in the ovs restart

gdyuldin commented on mos_tests/neutron/python_tests/test_dvr.py:
```diff
@@ -64,3 +65,180 @@ def test_vm_connectivity_wo_floating(self, variables):
 
         self.check_ping_from_vm(server, vm_keypair=self.instance_keypair)
+
+    def try_to_update_router_to_centralized(self, router_id):
```
Do you expect exception? If yes - let's use this check - https://pytest.org/latest/assert.html#assertions-about-expected-exceptions

TregubovYury replied:
mmm totally forgot that such frameworks usually have the expected exception feature

TimurNurlygayanov replied:
ok let's use them.

gdyuldin commented on mos_tests/neutron/python_tests/test_dvr.py:
```diff
@@ -64,3 +65,180 @@ def test_vm_connectivity_wo_floating(self, variables):
+        self.try_to_update_router_to_centralized(router['id'])
+
+    @pytest.fixture
+    def config_legacy_router(self, init):
```
Only router?

TregubovYury replied:
Yes, only router is created in this test, according to the original scenario.

gdyuldin replied:
But in method called `config_legacy_router` creates networks, instances, etc...

TregubovYury replied:
this test is for QA-311 QA-312
config_legacy_router is for QA-313 QA-315

rpromyshlennikov committed:
Add test network not on two dhcp agents .

rpromyshlennikov commented on mos_tests/neutron/python_tests/test_dvr.py:
```diff
@@ -17,6 +17,7 @@
 import pytest
 
 from mos_tests.neutron.python_tests import base
+from neutronclient.common.exceptions import NeutronClientException
```
should be:
 import pytest

 from neutronclient.common.exceptions import NeutronClientException

 from mos_tests.neutron.python_tests import base

Commit:
applied commits for pull #86

Author: Tregubov Yury

Commit:
applied commits for pull #86

Author: Tregubov Yury

Commit:
Added test for dvr router rescheduling

Author: Tregubov Yury

Added tests for QA-287 QA-288

Commit:
Added test for dvr router rescheduling

Author: Tregubov Yury

Added tests for QA-287 QA-288

Commit:
Merge pull request #90 from tkuterina/master

added TestOVSRestartAddFlows

Commit:
Merge pull request #82 from rpromyshlennikov/feature/QA-429-Test-not-two-agents-for-net

Feature/qa 429 test not two agents for net

Commit:
Merge branch 'master' of https://github.com/TregubovYury/mos-integration-tests

Conflicts:
	mos_tests/neutron/python_tests/test_dvr.py

Commit:
fixed some spell errors

Author: Tregubov Yury

fixed some spell errors and removed test which is not ready yet

TimurNurlygayanov commented on mos_tests/neutron/python_tests/test_dvr.py:
```diff
@@ -253,3 +257,247 @@ def test_routing_after_reset_computes(self, prepare_openstack, env_name):
+    def test_reschedule_router_from_snat_controller(self,
+                                                    create_dvr_router,
+                                                    create_net_and_vm):
+        """[Neutron DVR] Manually reschedule router from snat controller
```
Manually  > let's remove manually from here :)

TregubovYury commented:
After applying of comments from pull#96 need to re-base with fresh master. will create new pull request to keep the history clear.
